src/ring/middleware/basic_authentication.clj

```diff
@@ -18,18 +18,18 @@
   was raised."
   [direction-fn string]
   (try
-    (s/join (map char (direction-fn (.getBytes string))))
+    (s/join (map char (direction-fn (.getBytes ^String string))))
     (catch Exception _)))
 
 (defn- encode-base64
   "Will do a base64 encoding of a string and return a string."
   [^String string]
-  (byte-transform #(.encode (Base64/getEncoder) %) string))
+  (byte-transform #(.encode ^java.util.Base64$Encoder (Base64/getEncoder) ^bytes %) string))
```

Comment:
Omitting `^java.util.Base64$Encoder` does not yield reflection warning.  Please remove when not needed.